index.bs

```diff
@@ -1744,18 +1810,18 @@ interface MLGraph {
-                                            1. If outputs|[|outputName|].{{MLOutput/data}} is not an {{ArrayBufferView}}, then [=reject=] |promise| with an {{TypeError}} and stop.
-                                            1. If the kind of |outputs|[|outputName|].{{MLOutput/data}} is not compatible to output tensor according to [this table](#appendices-mloperandtype-arraybufferview-compatibility), then [=reject=] |promise| with a {{TypeError}} and stop.
-                                            1. If the length of |outputs|[|outputName|].{{MLOutput/data}} is less than |outputSize|, then [=reject=] |promise| with a {{TypeError}} and stop.
-                                            1. Set the content of |outputs|[|outputName|].{{MLOutput/data}} to the content of output tensor returned by |this|.{{MLGraph/[[implementation]]}}.
+                                        1. If |outputs|[|outputName|].{{MLTensor/resource}} was given, then:
+                                            1. If outputs|[|outputName|].{{MLTensor/resource}} is not an {{ArrayBufferView}}, then stop.
+                                            1. If the kind of |outputs|[|outputName|].{{MLTensor/resource}} is not compatible to output tensor according to [this table](#appendices-mloperandtype-arraybufferview-compatibility), then stop.
+                                            1. If the length of |outputs|[|outputName|].{{MLTensor/resource}} is less than |outputSize|, then stop.
+                                            1. Set the content of |outputs|[|outputName|].{{MLTensor/resource}} to the content of output tensor returned by |this|.{{MLGraph/[[implementation]]}}.
                                         1. Else:
-                                            1. Let |results|[|outputName|].{{MLOutput/data}} be a new {{ArrayBufferView}} of size |outputSize| and kind that is compatible to output tensor according to [this table](#appendices-mloperandtype-arraybufferview-compatibility).
-                                            1. Set the content of |results|[|outputName|].{{MLOutput/data}} to the content of output tensor returned by |this|.{{MLGraph/[[implementation]]}}.
+                                            1. Let |results|[|outputName|].{{MLTensor/resource}} be a new {{ArrayBufferView}} of size |outputSize| and kind that is compatible to output tensor according to [this table](#appendices-mloperandtype-arraybufferview-compatibility).
+                                            1. Set the content of |results|[|outputName|].{{MLTensor/resource}} to the content of output tensor returned by |this|.{{MLGraph/[[implementation]]}}.
                                     1. Remove |outputName| from |remainingOutputNames|.
-                                    1. If |remainingOutputNames| is empty, then resolve |promise| with |results| and stop.
+                                    1. If |remainingOutputNames| is empty, then stop.
```

Comment:
I think L1828 should be updated to either:

1. Return |results|.

Or remove this step if the preferred design is to use `data()` for data download per https://github.com/webmachinelearning/webnn/pull/162#discussion_r613776751